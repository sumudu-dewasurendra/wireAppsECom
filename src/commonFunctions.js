function calculateTotal(basketItems) {
  return basketItems.reduce((acc, item) => {
    return acc + item.product.price.amount * item.quantity;
  }, 0);
}

export function totalWithCurrency(basketItems) {
    if (basketItems.length > 0) {
        const total = calculateTotal(basketItems);
        const currency = basketItems?.[0].product.price.currency;
        return `${total} ${currency}`;
    }
    return 0
}