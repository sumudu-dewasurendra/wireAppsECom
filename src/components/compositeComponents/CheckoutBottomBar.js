import React from 'react';
import { Text, View } from 'react-native';
import Button from '../Button';
import { BUTTON_TYPES } from '../../utils/types';

const CheckoutBottomBar = ({
  total,
  onPress,
}) => {

  return (
    <View className={"h-[70px] absolute bottom-0 w-screen p-4 flex-row justify-between"}>
        <Text className={"text-2xl text-white font-bold"}>
          Total: <Text className={"text-secondary"}>{`${total}`}</Text>
        </Text>
        <Button
          color={"secondary"}
          textColor={"white"}
          label={"Checkout"}
          type={BUTTON_TYPES.SOLID}
          onPress={onPress}
        />
    </View>
  );
}

export default CheckoutBottomBar;