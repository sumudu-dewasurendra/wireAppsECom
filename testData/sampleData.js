export const productItems = [
    {
        "SKU":"1210",
        "brandName":"Nike",
        "colour":"blue",
        "description":"Hit the tracks in these Nike Air Relentless 4 featuring flexible forefoot sole and Reslon midsole underfoot cushioning for superior comfort at each step. The ridged outsole ensures excellent traction while the cushioned ankle collar and the anatomically shaped insole guarantee great support for the whole foot. The mesh upper panels provide breathability and airflow within the shoe.",
        "id":"1",
        "mainImage":"https://s3-eu-west-1.amazonaws.com/api.themeshplatform.com/media/7e386191b2ee40b290886a05d3e10e24_nike-air-relentless-a.jpg",
        "name":"Nike Air Relentless 4 Mens Running Shoes",
        "price":{
            "currency":"GBP",
            "amount":"45.00"
        },
        "sizes":['8','9','10','11'],
        "stockStatus":"IN STOCK"
    },
    {
        "SKU":"1217",
        "brandName":"Puma",
        "colour":"black",
        "description":"Originally the running shoe of 1986, the PUMA RS100 Opulence features a retro running look that is a blast from the past. Featuring R-System cushioning for excellent impact protection, this shoe has a leather, nubuck, and nylon upper and a polyurethane midsole. The rubber outsole increases durability. Includes reflective shoe laces.",
        "id":"5",
        "mainImage":"https://s3-eu-west-1.amazonaws.com/api.themeshplatform.com/media/86d1c3b3500645a3bb1312f945a9cf38_35686403_fr_puma_sc7.jpeg",
        "name":"PUMA RS100 Opulence",
        "price":{
            "amount":"45.00",
            "currency":"GBP"
        },
        "sizes":['8','9','10','11'],
        "stockStatus":"IN STOCK"
    }
]

export const basketItems = [
    {
        "product": {
            "SKU":"1210",
            "brandName":"Nike",
            "colour":"blue",
            "description":"Hit the tracks in these Nike Air Relentless 4 featuring flexible forefoot sole and Reslon midsole underfoot cushioning for superior comfort at each step. The ridged outsole ensures excellent traction while the cushioned ankle collar and the anatomically shaped insole guarantee great support for the whole foot. The mesh upper panels provide breathability and airflow within the shoe.",
            "id":"1",
            "mainImage":"https://s3-eu-west-1.amazonaws.com/api.themeshplatform.com/media/7e386191b2ee40b290886a05d3e10e24_nike-air-relentless-a.jpg",
            "name":"Nike Air Relentless 4 Mens Running Shoes",
            "price":{
                "currency":"GBP",
                "amount":"45.00"
            },
            "sizes":['8','9','10','11'],
            "stockStatus":"IN STOCK"
        },
        "quantity": 1,
        "size": "8",
    },
    {
        "product": {
            "SKU":"1217",
            "brandName":"Puma",
            "colour":"black",
            "description":"Originally the running shoe of 1986, the PUMA RS100 Opulence features a retro running look that is a blast from the past. Featuring R-System cushioning for excellent impact protection, this shoe has a leather, nubuck, and nylon upper and a polyurethane midsole. The rubber outsole increases durability. Includes reflective shoe laces.",
            "id":"5",
            "mainImage":"https://s3-eu-west-1.amazonaws.com/api.themeshplatform.com/media/86d1c3b3500645a3bb1312f945a9cf38_35686403_fr_puma_sc7.jpeg",
            "name":"PUMA RS100 Opulence",
            "price":{
                "amount":"45.00",
                "currency":"GBP"
            },
            "sizes":['8','9','10','11'],
            "stockStatus":"IN STOCK"
        },
        "quantity": 1,
        "size": "8",
    }
]

