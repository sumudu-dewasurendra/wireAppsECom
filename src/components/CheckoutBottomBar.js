import React from 'react';
import { Text, View, TouchableOpacity } from 'react-native';
import { totalWithCurrency } from '../common/commonFunctions';

const CheckoutBottomBar = ({
  basketItems,
}) => {

  return (
    <View className={"h-[70px] absolute bottom-0 w-screen p-4 flex-row justify-between"}>
        <Text className={"text-2xl text-white font-bold"}>{`Total: ${totalWithCurrency(basketItems)}`}</Text>
        <TouchableOpacity className={"bg-secondary justify-center items-center p-2 rounded-lg"}>
            <Text className={"text-white text-1xl rounded-lg font-bold "}>Checkout</Text>
        </TouchableOpacity>
    </View>
  );
}

export default CheckoutBottomBar;