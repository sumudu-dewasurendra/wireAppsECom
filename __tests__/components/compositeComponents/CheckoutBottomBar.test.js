import React from 'react';
import renderer from 'react-test-renderer';
import CheckoutBottomBar from '../../../src/components/compositeComponents/CheckoutBottomBar';
import { basketItems } from '../../../testData/sampleData';
import { totalWithCurrency } from '../../../src/common/commonFunctions';

test('renders correctly', () => {
  const tree = renderer.create(<CheckoutBottomBar
    total={totalWithCurrency(basketItems)} />).toJSON();
  expect(tree).toMatchSnapshot();
});