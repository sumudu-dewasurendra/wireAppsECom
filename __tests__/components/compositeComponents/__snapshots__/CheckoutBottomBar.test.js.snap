// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "height": 70,
      },
      {
        "position": "absolute",
      },
      {
        "bottom": 0,
      },
      {
        "width": 750,
      },
      {
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      {
        "flexDirection": "row",
      },
      {
        "justifyContent": "space-between",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 24,
          "lineHeight": 32,
        },
        {
          "color": "#fff",
        },
        {
          "fontWeight": "700",
        },
      ]
    }
  >
    Total: 
    <Text
      style={
        [
          {
            "color": "#ff8626",
          },
        ]
      }
    >
      90 GBP
    </Text>
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#fff",
          },
          {
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
          },
          {
            "fontWeight": "700",
          },
        ]
      }
    >
      Checkout
    </Text>
  </View>
</View>
`;
